### PR TITLE
implemented Saving feature

### PR DESCRIPTION
**Pull request by passionfruitjello**

### CODE REVIEW

File: src/main/java/save/Save.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package save;
+
+import java.util.Map;
+
+public class Save {
+    Map<Id, SaveableEntity> savedData;
```
Comment (bw55555) [inline src/main/java/save/Save.java:6]: Should be Map<Id, String>

File: src/main/java/save/Id.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
+package save;
+
+public enum Id {
+    A {
+        public String toString() {
```
Comment (bw55555) [inline src/main/java/save/Id.java:5]: Recommend to use the builtin toString and valueOf methods
that way you don't have to use 1 letter ids and can use more descriptive ones

File: src/main/java/save/SaveManager.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
+package save;
+
+import java.util.ArrayList;
+
+
+public class SaveManager {
+    ArrayList<SaveableEntity> entities;
+    ArrayList<Save> saves;
+    SaveLoader loader = new SaveLoader();
```
Comment (bw55555) [inline src/main/java/save/SaveManager.java:9]: by convention, should move object initialization into constructor

File: src/main/java/save/SaveFactory.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+    /**
+     * @param entities the list of Saveable entities
+     * @return a Save created
+     */
+    public Save createSave(List<SaveableEntity> entities) {
+        Map<Id, SaveableEntity> dataToSave = new HashMap<Id, SaveableEntity>();
+        int size = entities.size();
+        int i = 0;
+        for (Id id : Id.values()) {
```
Comment (bw55555) [inline src/main/java/save/SaveFactory.java:18]: here should iterate on entities, not id.

**Review by bw55555 — CHANGES_REQUESTED** [review]

**Review by Yasutakahi — COMMENTED** [review]
If you are submitting this as a pull request for the milestone 6, make sure you write the description later, so TA can know what is happening in the pull request

**Review by bw55555 — APPROVED** [review]
Looks good now!

**Review by marjaleenakiik — APPROVED** [review]
Includes suitable documentation.  Code functions as intended.

**Review by bw55555 — APPROVED** [review]

**Review by bw55555 — COMMENTED** [review]
Rename to not override default toString() method

**Review by marjaleenakiik — APPROVED** [review]
See previous comment, small change from before.

**Review by juhwansonn — APPROVED** [review]

**Pull request merged**